### PR TITLE
fix(redirect): redirect old links to the new ones

### PR DESCRIPTION
**Pull request by xMohamd**
## Description

This PR updated `redirect.json`  

## Validation

[https://nodejs.org/en/docs/guides/debugging-getting-started/](https://nodejs.org/en/docs/guides/debugging-getting-started/) => [https://nodejs.org/en/learn/getting-started/debugging](https://nodejs.org/en/learn/getting-started/debugging)

[https://nodejs.org/en/docs/guides/working-with-different-filesystems/](https://nodejs.org/en/docs/guides/working-with-different-filesystems/) => [https://nodejs.org/en/learn/manipulating-files/working-with-different-filesystems](https://nodejs.org/en/learn/manipulating-files/working-with-different-filesystems)

[https://nodejs.org/en/docs/guides/anatomy-of-an-http-transaction/ ](https://nodejs.org/en/docs/guides/anatomy-of-an-http-transaction/ ) => [https://nodejs.org/en/learn/modules/anatomy-of-an-http-transaction](https://nodejs.org/en/learn/modules/anatomy-of-an-http-transaction)

[https://nodejs.org/en/docs/guides/publishing-napi-modules/](https://nodejs.org/en/docs/guides/publishing-napi-modules/) => [https://nodejs.org/en/learn/modules/publishing-node-api-modules](https://nodejs.org/en/learn/modules/publishing-node-api-modules)

## Related Issues

#6338 

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npx turbo lint` to ensure the code follows the style guide. And run `npx turbo lint:fix` to fix the style errors if necessary.
- [x] I have run `npx turbo format` to ensure the code follows the style guide.
- [x] I have run `npx turbo test` to check if all tests are passing.
- [ ] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

File: redirects.json [hunk 1]
```diff
@@ -180,6 +180,22 @@
       "source": "/:locale/docs/guides/timers-in-node/",
       "destination": "/:locale/learn/asynchronous-work/discover-javascript-timers"
     },
+    {
+      "source": "/:locale/docs/guides/debugging-getting-started/",
+      "destination": "/:locale/learn/getting-started/debugging"
+    },
```
Comment (richardlau) [inline redirects.json]: This is already redirected earlier in the file (via https://github.com/nodejs/nodejs.org/pull/6290).

Comment (xMohamd) [reply redirects.json]: Ok, I'll remove it but i'll wait if there is any other comments

Comment (ovflowd) [reply redirects.json]: You can remove it as Richard mentioned :)

File: redirects.json [hunk 2]
```diff
@@ -180,6 +180,18 @@
       "source": "/:locale/docs/guides/timers-in-node/",
       "destination": "/:locale/learn/asynchronous-work/discover-javascript-timers"
     },
+    {
+      "source": "/:locale/docs/guides/working-with-different-filesystems/",
```
Comment (ovflowd) [inline redirects.json]: BTW there shouldn't be /docs/ on any of these links.

**Review by ovflowd — APPROVED** [review]
LGTM!

**Pull request merged**